mingw-llvm error non-constant-expression cannot be narrowed from type 'DWORD' (aka 'unsigned long') to 'int' in initializer list [-Wc++11-narrowing]

### Tested versions

- Reproducable in https://github.com/godotengine/godot/commit/9ab5cedef6de32826b9184e960b3880df3888a5a
- Not reproducable in https://github.com/godotengine/godot/releases/tag/4.2.1-stable

### System information

Windows 11, Nvidia 3090

### Issue description

```
platform\windows\display_server_windows.cpp:250:4: error: non-constant-expression cannot be narrowed from type 'DWORD' (aka 'unsigned long') to 'int' in initializer list [-Wc++11-narrowing]
  250 |                         QITABENT(FileDialogEventHandler, IFileDialogEvents),
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1198:31: note: expanded from macro 'QITABENT'
 1198 | #define QITABENT(Cthis, Ifoo) QITABENTMULTI(Cthis, Ifoo, Ifoo)
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1193:62: note: expanded from macro 'QITABENTMULTI'
 1193 | #define QITABENTMULTI(Cthis, Ifoo, Iimpl) { &__uuidof(Ifoo), OFFSETOFCLASS(Iimpl, Cthis) }
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1189:38: note: expanded from macro 'OFFSETOFCLASS'
 1189 | #define OFFSETOFCLASS(base, derived) ((DWORD)(DWORD_PTR)(static_cast<base*>((derived*)8))-8)
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
platform\windows\display_server_windows.cpp:250:4: note: insert an explicit cast to silence this issue
  250 |                         QITABENT(FileDialogEventHandler, IFileDialogEvents),
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1198:31: note: expanded from macro 'QITABENT'
 1198 | #define QITABENT(Cthis, Ifoo) QITABENTMULTI(Cthis, Ifoo, Ifoo)
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1193:62: note: expanded from macro 'QITABENTMULTI'
 1193 | #define QITABENTMULTI(Cthis, Ifoo, Iimpl) { &__uuidof(Ifoo), OFFSETOFCLASS(Iimpl, Cthis) }
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1189:38: note: expanded from macro 'OFFSETOFCLASS'
 1189 | #define OFFSETOFCLASS(base, derived) ((DWORD)(DWORD_PTR)(static_cast<base*>((derived*)8))-8)
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
platform\windows\display_server_windows.cpp:251:4: error: non-constant-expression cannot be narrowed from type 'DWORD' (aka 'unsigned long') to 'int' in initializer list [-Wc++11-narrowing]
  251 |                         QITABENT(FileDialogEventHandler, IFileDialogControlEvents),
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1198:31: note: expanded from macro 'QITABENT'
 1198 | #define QITABENT(Cthis, Ifoo) QITABENTMULTI(Cthis, Ifoo, Ifoo)
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1193:62: note: expanded from macro 'QITABENTMULTI'
 1193 | #define QITABENTMULTI(Cthis, Ifoo, Iimpl) { &__uuidof(Ifoo), OFFSETOFCLASS(Iimpl, Cthis) }
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1189:38: note: expanded from macro 'OFFSETOFCLASS'
 1189 | #define OFFSETOFCLASS(base, derived) ((DWORD)(DWORD_PTR)(static_cast<base*>((derived*)8))-8)
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
platform\windows\display_server_windows.cpp:251:4: note: insert an explicit cast to silence this issue
  251 |                         QITABENT(FileDialogEventHandler, IFileDialogControlEvents),
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1198:31: note: expanded from macro 'QITABENT'
 1198 | #define QITABENT(Cthis, Ifoo) QITABENTMULTI(Cthis, Ifoo, Ifoo)
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1193:62: note: expanded from macro 'QITABENTMULTI'
 1193 | #define QITABENTMULTI(Cthis, Ifoo, Iimpl) { &__uuidof(Ifoo), OFFSETOFCLASS(Iimpl, Cthis) }
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ernest.lee/scoop/apps/llvm-mingw/current/include/shlwapi.h:1189:38: note: expanded from macro 'OFFSETOFCLASS'
 1189 | #define OFFSETOFCLASS(base, derived) ((DWORD)(DWORD_PTR)(static_cast<base*>((derived*)8))-8)
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
```

### Steps to reproduce

scons werror=yes dev_build=no precision=double compiledb=yes target=editor tests=yes debug_symbols=yes

### Minimal reproduction project (MRP)

N/A

--- a/platform/windows/display_server_windows.cpp
+++ b/platform/windows/display_server_windows.cpp
@@ -247,8 +247,13 @@ public:
 	// IUnknown methods
 	HRESULT STDMETHODCALLTYPE QueryInterface(REFIID riid, void **ppv) {
 		static const QITAB qit[] = {
+#ifdef __MINGW32__
+			{ &__uuidof(IFileDialogEvents), static_cast<decltype(qit[0].dwOffset)>(OFFSETOFCLASS(IFileDialogEvents, FileDialogEventHandler)) },
+			{ &__uuidof(IFileDialogControlEvents), static_cast<decltype(qit[0].dwOffset)>(OFFSETOFCLASS(IFileDialogControlEvents, FileDialogEventHandler)) },
+#else
 			QITABENT(FileDialogEventHandler, IFileDialogEvents),
 			QITABENT(FileDialogEventHandler, IFileDialogControlEvents),
+#endif
 			{ 0, 0 },
 		};
 		return QISearch(this, qit, riid, ppv);

